File: BocetoV2/Clases/EstadoAnimo.java
import java.time.LocalDate;
import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public class EstadoAnimo {
    private Map<LocalDate, IconoEstadoAnimo> registroAnimo;

    public EstadoAnimo() {
        this.registroAnimo = new HashMap<>();
    }

    public Map<LocalDate, IconoEstadoAnimo> getRegistroAnimo() { return registroAnimo; }
    public void setRegistroAnimo(Map<LocalDate, IconoEstadoAnimo> registroAnimo) { this.registroAnimo = registroAnimo; }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EstadoAnimo that = (EstadoAnimo) o;
        return Objects.equals(registroAnimo, that.registroAnimo);
    }

    @Override
    public int hashCode() { return Objects.hash(registroAnimo); }

    @Override
    public String toString() { return "EstadoAnimo{" + "registros=" + registroAnimo.size() + '}'; }
}

File: App/AppV1/app/src/main/java/com/pmdm/appV1/models/EstadoAnimo.kt
package com.pmdm.appV1.models

import java.time.LocalDate

data class EstadoAnimo(
    // En Kotlin usamos MutableMap para indicar que se puede modificar
    var registroAnimo: MutableMap<LocalDate, IconoEstadoAnimo> = mutableMapOf()
) {
    override fun toString(): String {
        return "EstadoAnimo(registros=${registroAnimo.size})"
    }
}
